### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by rgreenblatt**
Main 5.8 pr. Specifically solves:
https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/502
https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/499
https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/484
https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/501
https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/490

Implements part of:
CrossTheRoadElectronics/Phoenix-IssueTracker#475
CrossTheRoadElectronics/Phoenix-IssueTracker#488
CrossTheRoadElectronics/Phoenix-IssueTracker#486

### CODE REVIEW

**Comment by rgreenblatt** [comment]
Also now solves https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/497 / https://github.com/CrossTheRoadElec/Phoenix-frc-lib/issues/18 

tested here: https://github.com/CrossTheRoadElec/Phoenix-frc-lib/pull/55/commits/d56cfbecf7c8f634a0c96edeea3df68a0f8c3a33#diff-29661677fb334a4d55ce2ce9df8cccccR141
(tested that problem reproduces without change and change solves it)

**Comment by rgreenblatt** [comment]
In this branch, all hal reporting calls are commented out in java. ~This needs to be resolved before release.~

**Comment by rgreenblatt** [comment]
This pr solves https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/492 and https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/477

(I forgot to mention this earlier)

File: src/main/native/cpp/HsvToRgb.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#ifndef CTR_EXCLUDE_WPILIB_CLASSES
+#if defined(CTR_INCLUDE_WPILIB_CLASSES) || defined(__FRC_ROBORIO__)
```
Comment (ozrien) [inline src/main/native/cpp/HsvToRgb.cpp]: I'm not sure why this ifdef is here.  This class shouldn't be impacted by RIO

File: src/main/native/cpp/RCRadio3Ch.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#ifndef CTR_EXCLUDE_WPILIB_CLASSES
+#if defined(CTR_INCLUDE_WPILIB_CLASSES) || defined(__FRC_ROBORIO__)
```
Comment (ozrien) [inline src/main/native/cpp/RCRadio3Ch.cpp]: I'm not sure why this ifdef is here.  This class shouldn't be impacted by RIO

**Pull request closed without merging**